iceoryx_binding_c/source/c_publisher.cpp

```diff
@@ -30,25 +31,63 @@ extern "C" {
 #include "iceoryx_binding_c/publisher.h"
 }
 
+static uint64_t PUBLISHER_OPTIONS_INIT_CHECK_CONSTANT = 123454321;
+
+void iox_pub_options_init(iox_pub_options_t* options)
+{
+    if (options == nullptr)
+    {
+        LogWarn() << "publisher options initialization skipped - null pointer provided";
+        return;
+    }
+
+    PublisherOptions publisherOptions;
+    options->historyCapacity = publisherOptions.historyCapacity;
+    options->nodeName = nullptr;
+
+    options->initCheck = PUBLISHER_OPTIONS_INIT_CHECK_CONSTANT;
+}
+
+bool iox_pub_options_is_initialized(const iox_pub_options_t* const options)
+{
+    if (options && options->initCheck == PUBLISHER_OPTIONS_INIT_CHECK_CONSTANT)
+    {
+        return true;
+    }
+    return false;
```

Comment:
I've seen this in Walter E. Browns Meeting C++ 2019 talk "multiple ways to return true/false" ;)

Reply:
Not exactly but yes, can be shortened. There are other points in the code where I opted not to use e.g. the ternary operator for readability.